src/cpp/src/sequence_group.cpp

```diff
@@ -22,22 +22,78 @@ size_t Sequence::_make_hash(size_t content_length) {
+                content.insert(content.end(), prompt_ids.begin() + block_start_idx, prompt_ids.begin() + std::min(prompt_ids.size(), content_length));
+            }
+            if (content_length > prompt_ids.size()) {
+                size_t start = block_start_idx < prompt_ids.size() ? 0 : block_start_idx - prompt_ids.size();
+                content.insert(content.end(), m_generated_ids.begin() + start, m_generated_ids.begin() + content_length - prompt_ids.size());
+            }
+            data = reinterpret_cast<char*>(content.data());
+            size = content.size() * sizeof(content[0]);
         }
-        if (content_length > prompt_ids.size()) {
-            size_t start = block_start_idx < prompt_ids.size() ? 0 : block_start_idx - prompt_ids.size();
-            content.insert(content.end(), m_generated_ids.begin() + start, m_generated_ids.begin() + content_length - prompt_ids.size());
+        else if (sequence_group->get_sequence_group_type() == SequenceGroupType::EMBEDDINGS) {
+            const auto input_embeds = sequence_group->get_input_embeds();
+            const auto generated_embeds = m_generated_ids_embeds;
+            OPENVINO_ASSERT(content_length <= input_embeds.size() + generated_embeds.size());
+            std::vector<float> content_float;
+
+            // get inputs embeddings
+            if (block_start_idx < input_embeds.size()) {
+                for (size_t idx = block_start_idx; idx < std::min(input_embeds.size(), content_length); idx++) {
+                    auto embed = _reduce_embedding(input_embeds[idx]);
+                    const char* embed_char = reinterpret_cast<const char*>(embed.data());
+                    content_float.insert(content_float.end(), embed.begin(), embed.end());
+                }
+            }
+
+            // get generated ids embeddings
+            if (content_length > input_embeds.size()) {
+                size_t start = block_start_idx < input_embeds.size() ? 0 : block_start_idx - input_embeds.size();
+                for (size_t idx = start; idx < content_length - input_embeds.size(); idx++) {
+                    auto embed = _reduce_embedding(generated_embeds[idx]);
+                    content_float.insert(content_float.end(), embed.begin(), embed.end());
+                }
+            }
+
+            size_t prev_hashes_size = content.size() == 0 ? 0 : content.size() * sizeof(content[0]);
+            size_t content_float_size = content_float.size() * sizeof(content_float[0]);
+            size = prev_hashes_size + content_float_size;
+            data = new char[size];
+            
+            // append previously calculated prefix hashes if they are available
+            if (prev_hashes_size) {
+                auto prev_hashes = reinterpret_cast<const char*>(content.data());
+                std::copy_n(prev_hashes, prev_hashes_size, data);
+            }
+
+            auto content_char = reinterpret_cast<const char*>(content_float.data());
```

Comment:
maybe instead of interpretation floats as bytes, we can convert them to integers via multiply to some value and round to int? 
It will hide some deviations in FP math if they happen.